src/api/admin.ts

```diff
@@ -158,7 +159,10 @@ app.patch('/providerKeys/:id', (req, res) => {
 })
 
 app.get('/providers', async (_, res) => {
-  const data = await Promise.all([providers.dns_gd.getDomains()])
+  const data = await Promise.all([
+    providers.dns_gd.getDomains(),
+    providers.dns_namecom.getDomains()
```

Comment:
I would prefer to have new providers not hhave to add any code in here. Could we change the logic here to use a `filter` and then `Promise.all`? This way, new providers don't have to update this when the want thheir service to be supported.

Reply:
Sure I was thinking about to do a generic function that look at `src/api/serviceConfig.ts` or even maybe a config file with a list of all providers that work with `acme.sh` then will we just have to uncomment provider to enable it.
```
[
  dns_gd,
  dns_namecom,
  // dns_provider,
 //  ... and so on
]
```

Reply:
or this:

```
const requests = Object.keys(providers).filter((name) => {
  return providers[name].getDomains
}).map( (name) => {
  return providers[name].getDomains()
})

const data = await Promise.all( requests )
```